Src/VimCore/CoreInterfaces.fs

```diff
@@ -4300,6 +4300,12 @@ and IVimBufferData =
     /// The current directory for this particular window
     abstract CurrentDirectory: string option with get, set
 
+    /// The current (rooted) file path for this buffer
+    abstract CurrentFilePath : string
+
+    /// The current file name for this buffer. This is CurrentFilePath, excluding any prefix shared with CurrentDirectory.
+    abstract CurrentFileName : string

```

Comment:
These should both return `string option`. Not all instances of `IVimBufferData` will end up having a path. 

Still thinking about whether this is the right place to expose file paths. Up until now its mostly been hidden behind `IVimHost` and methods like `Save`. Think this is probably the right place though.

Reply:
@jaredpar , do you know of steps to create a buffer without a file path in VS?  Although it makes logical sense to change the properties to `string option`, any newly created document in VS seems to have a backing file in the temp directory, e.g. `~vs6503.cs`.  What's worse is that despite significant time spent searching through VS extensibility docs, I can't find any supported way of determining whether not a file has been saved at least once.  I suppose I could check whether the file is in a known temp directory and matches the '~vs' pattern, but this is isn't exactly bulletproof.

Let me know if you decide you want me to rewrite the `IVimBufferData.CurrentFile*` properties as `IVim` or `IVimHost` functions.  Either would be an easy change as both interfaces are accessible via `IVimBufferData` properties.  I don't think I realized early on that they were just a 'dot' or two away.

Reply:
Re: CurrentRelativeFileName, I agree we should probably favor .NET's naming conventions over Vim's.  Vim calls the '%' and '#' symbolic names "filename", but this actually caused me a lot of confusion until I figured out that it's not always just the file name -- it's the full file path minus any common prefix it shares with the current directory.  However, CurrentFilePathWithoutCurrentDirectorySharedPrefix is a bit absurd, so CurrentRelativeFileName it is!

Reply:
> do you know of steps to create a buffer without a file path in VS?

There are a couple of examples of such buffers. Most notably is the `ITextBuffer` that is created to edit conditional break points in managed code. 

> any newly created document in VS seems to have a backing file in the temp directory, e.g.

Indeed that will be the case for documents, or rather anything that can be saved, in VS. Their implementation is very tied to the file system and it always has a path in that case.